### PR TITLE
Feature/refactor maintenance

### PR DESCRIPTION
**Pull request by nqb**
# Description
- Build packages using Makefile (in place of commands in `.gitlab-ci.yml`)
- Extract release version from `conf/pf-release` to use it during several stages (build_docker_img, test, upload, build_pf_img)
- Use all runners for upload stage
- Upload packages directly to web.inverse.ca (no more pfbuilder)
- Use a RPM and Debian repository per-version (see #6162)
- New pristine Debian 9 pristine box (to test from scratch, see #6292)
- Add a script to simplify release process (specially for patch release)
- New names for SQL schema upgrade scripts (X.Y.Z => X.Y)
- Always install packetfence packages from local repository in pipeline
- Build sources using during RPM packaging with `tar` in place of `git archive` to include untracked file (like `git_commit_id`)

Additional (related to RHEL8 and Debian 11):
- Add Vagrant boxes for each distribution

Deprecations:
- `pf-maint.pl `
- `packetfence-devel` and `debian-devel` repositories

Notes:
* Documentation will be updated on PR related to RHEL8 and Debian 11.
* Build of Docker images is not touched because we will start to use new Docker images for RHEL8 and Debian 11

# Dependencies
- https://github.com/inverse-inc/ansible-packetfence/pull/34 (merged)

# Impacts
- Package builds for stable, devel and branches environment
- Maintenance patches 

# Delete branch after merge
YES

# Checklist
- [x] (WIP) Document the feature
- [x] Post-merge actions

# NEWS file entries
## Enhancements
* Get maintenance patches through package manager

### CODE REVIEW

File: addons/vagrant/requirements.yml [hunk 1]
```diff
@@ -3,12 +3,24 @@ roles:
   - src: geerlingguy.nodejs
   - src: inverse_inc.gitlab_buildpkg_tools
 
+# to test locally with Vagrant (due to --force option)
```
Comment (JeGoi) [inline addons/vagrant/requirements.yml]:
```suggestion
# For roles, to test locally with Vagrant (due to --force option)
```

File: addons/vagrant/requirements.yml [hunk 2]
```diff
@@ -3,12 +3,24 @@ roles:
...
   - name: inverse_inc.packetfence
+    version: 1.2.0-1
   - name: debops.debops
   - name: inverse_inc.windows
   - name: inverse_inc.cumulus
   - name: inverse_inc.utils
   - name: inverse_inc.wireless
 
-
+# to test locally with Vagrant (due to --force option)
```
Comment (JeGoi) [inline addons/vagrant/requirements.yml]:
```suggestion
# For collections, to test locally with Vagrant (due to --force option)
```

File: ci/lib/release/prep-release.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+DEB_CHLOG=${DEB_DIR}/changelog
+
+configure_and_check() {
+    log_subsection "Get current PacketFence version"    
+    echo "${PF_PATCH_RELEASE}"
+    log_subsection "Get new PacketFence version"
+    read -p "New PacketFence version: " PF_NEW_PATCH_RELEASE
+}
+
+update_pf_version() {
```
Comment (JeGoi) [inline ci/lib/release/prep-release.sh:27]: Is there anything specific to do on Debian side?

File: ci/lib/test/Makefile [hunk 4]
```diff
@@ -19,7 +19,7 @@ all: test-pfcen7dev-dev test-pfdeb9dev-dev test-pfcen7dev-release test-pfdeb9dev
 ### Targets used in pipeline
 .PHONY: test
 test: clean
-	./vagrant-wrapper.sh run
+	PF_MINOR_RELEASE=$(PF_MINOR_RELEASE) ./vagrant-wrapper.sh run
```
Comment (JeGoi) [inline ci/lib/test/Makefile:22]: So, no need to export, correct?

File: ci/lib/upload/deploy-artifacts.sh [hunk 5]
```diff
@@ -1,54 +1,74 @@
...
         else
             echo "Keeping SRPMS according to '$DEPLOY_SRPMS' value"
         fi
 
-        echo "copy: $src_dir/*.rpm -> $dst_dir/RPMS"
-        scp $src_dir/*.rpm $dst_dir/RPMS \
+        # copy rpm files
+        echo "copy: $src_dir/*.rpm -> $dst_dir"
+        scp $src_dir/*.rpm $dst_dir \
             || die "scp failed"
```
Comment (JeGoi) [inline ci/lib/upload/deploy-artifacts.sh:68]: Didn't you remove the die function (line 4)?

Comment (JeGoi) [reply ci/lib/upload/deploy-artifacts.sh:68]: They are several references of die function in this file.

File: ci/packer/zen/Makefile [hunk 6]
```diff
@@ -37,15 +41,15 @@ zen-release:
 	make \
 	PKR_VAR_pf_repo=packetfence \
 	PKR_VAR_pf_package=packetfence \
-	PKR_VAR_pf_branch=stable \
+	PKR_VAR_pf_release_pkg=packetfence-release-$(PF_MINOR_RELEASE)-el7.noarch.rpm \
```
Comment (JeGoi) [inline ci/packer/zen/Makefile:44]: What will happened for rhel8?

**Review by JeGoi — COMMENTED** [review]
Few questions but it looks good.
Nice job

File: ci/lib/release/prep-release.sh (hunk 3 above)
Comment (nqb) [reply ci/lib/release/prep-release.sh:27]: No.
In fact, purpose of this script is to update PacketFence source code before a release pipeline in order to have a valid build. So there is nothing more to do on Debian side.

File: ci/lib/test/Makefile (hunk 4 above)
Comment (nqb) [reply ci/lib/test/Makefile:22]: No. We pass `PF_MINOR_RELEASE` variable as an environment variable to `vagrant-wrapper.sh`

File: ci/lib/upload/deploy-artifacts.sh (hunk 5 above)
Comment (nqb) [reply ci/lib/upload/deploy-artifacts.sh:68]: Function is defined here: https://github.com/inverse-inc/packetfence/blob/8e214fd240d5d1b950c90083cf95429be9c420ef/ci/lib/common/functions.sh#L10-L12

Then `functions.sh` is sourced by `deploy_artifacts.sh` script:
https://github.com/inverse-inc/packetfence/blob/8e214fd240d5d1b950c90083cf95429be9c420ef/ci/lib/upload/deploy-artifacts.sh#L13

File: ci/packer/zen/Makefile (hunk 6 above)
Comment (nqb) [reply ci/packer/zen/Makefile:44]: Because we will only support one EL group at a time, I will replace el7 by el8 on `feature/rhel8` branch.

**Comment by nqb** [comment]
Commit https://github.com/inverse-inc/packetfence/pull/6378/commits/e1f0efb60e4b4a8ae3789b229e2a298291e87035 will break builds. I'm working on fix before we merge.

**Pull request merged**